### ISSUE TITLE
False Positive for DL3006 with arg-generated image names

### ISSUE THREAD

**Issue by JWCS** [hadolint/hadolint#979]
- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

Note: a subset of this edge case seems to be addressed in open PR #942

### Expected behavior
No warning on lines of `FROM base_${arg} AS base`

### Actual behavior
DL3006 error on that line, that there is no tag.
Error goes away if there's a tag, but that doesn't really make sense...

### Steps to reproduce the behavior
I ran this on the online linter. This is an example subset of production dockerfile, where two different sources, as well as any source specific setup, can be selected by an arg.

**Dockerfile** (if relevant)

```Dockerfile
ARG src=1 # {1, 2}
FROM src_loc/ex1:tag AS img1
FROM src_loc/ex2:tag AS img2
# Error: DL3006
FROM img${src} AS img_err
# No Error:
FROM img${src}:tag AS img_ok
# ...
```

**Additional environment details (OS, stack version, etc.)**
I ran this on the online linter, not locally

**Comment by roadSurfer** [comment]
I hit a similar issue today:

```
ARG BASE_IMAGE_NAME="registry/my-base:1.2.3"

# Triggers DL3006 warning
FROM ${BASE_IMAGE_NAME} AS builder
...

# Triggers DL3006 warning
FROM ${BASE_IMAGE_NAME}
WORKDIR /
COPY --from=builder blah-blah .
...
```

Is it just doing a basic check for `FROM this:that`? Can it not resolve the `ARG` values?
